src/ui/Omnibox/Omnibox.ts

```diff
@@ -842,6 +842,9 @@ export class Omnibox extends Component {
 
   private shouldExecuteQuery(searchAsYouType: boolean) {
     const text = this.getQuery(searchAsYouType);
+    if (this.searchInterface.options.allowQueriesWithoutKeywords === false) {
+      return this.lastQuery != text && Utils.isNonEmptyString(text);
+    }
     return this.lastQuery != text && text != null;
```

Comment:
I'm curious as to why we should execute the query when the query **is** an empty string and when `allowQueriesWithoutKeywords` is `true`? Shouldn't the non-empty string apply always?

Reply:
It's "isNonEmptyString", so if we do not allow queries without keywords, we should not execute the query if it is an empty string.

Reply:
Are you referring to line 848? On that line, we are checking `text != null`, so if text is an empty string, it will return `true`. Is this desired behaviour?